File: generate-og.py
#!/usr/bin/env python3
"""Generate OG image (1200x630 PNG) for PicEte."""
from PIL import Image, ImageDraw, ImageFont
import math, os

W, H = 1200, 630
img = Image.new('RGBA', (W, H), (0, 0, 0, 0))
draw = ImageDraw.Draw(img)

# Background gradient
for y in range(H):
    t = y / H
    r = int(15 + t * 15)  # 0x0f -> 0x1e
    g = int(23 + t * 20)
    b = int(42 + t * 17)
    draw.line([(0, y), (W, y)], fill=(r, g, b))

# Grid dots
for x in range(60, W, 120):
    for y in range(60, H, 120):
        draw.ellipse([x-2, y-2, x+2, y+2], fill=(51, 65, 85, 76))

# Decorative shapes
draw.rounded_rectangle([80, 80, 160, 160], radius=16, outline=(8, 173, 255, 38), width=1)
draw.rounded_rectangle([1040, 470, 1120, 550], radius=16, outline=(96, 165, 250, 38), width=1)
draw.arc([1080, 140, 1120, 180], 0, 360, fill=(8, 173, 255, 25), width=1)
draw.arc([86, 516, 114, 544], 0, 360, fill=(96, 165, 250, 25), width=1)

# Try to use system fonts
font_paths = [
    '/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf',
    '/usr/share/fonts/truetype/dejavu/DejaVuSans.ttf',
]
try:
    bold_font = ImageFont.truetype(font_paths[0], 60)
    reg_font = ImageFont.truetype(font_paths[1], 28)
    small_font = ImageFont.truetype(font_paths[1], 14)
    url_font = ImageFont.truetype(font_paths[0], 17)
except:
    bold_font = ImageFont.load_default()
    reg_font = bold_font
    small_font = bold_font
    url_font = bold_font

# "Pic" in light
draw.text((420, 180), "Pic", fill=(226, 232, 240), font=bold_font)
# "Ete" in blue
draw.text((540, 180), "Ete", fill=(8, 173, 255), font=bold_font)

# Tagline
draw.text((600, 320), "Free Online Image Processing Tools", fill=(148, 163, 184), font=reg_font, anchor='mt')

# Feature pills
pill_data = [
    ('Format Converter', 390),
    ('Image Resizer', 600),
    ('Compressor', 810),
]
pill_y = 400
for text, cx in pill_data:
    # pill background
    x0, y0 = cx - 65, pill_y
    x1, y1 = cx + 65, pill_y + 28
    draw.rounded_rectangle([x0, y0, x1, y1], radius=14, fill=(30, 41, 59), outline=(51, 65, 85), width=1)
    # text
    tw = draw.textlength(text, font=small_font)
    tx = cx - tw // 2
    draw.text((tx, pill_y + 5), text, fill=(203, 213, 225), font=small_font)

# URL
draw.text((600, 500), "picete.com", fill=(8, 173, 255), font=url_font, anchor='mt')

output_path = '/home/wu/picete-site/images/og-image.png'
img.save(output_path)
print(f'OG Image saved: {output_path} ({(os.path.getsize(output_path) / 1024):.1f} KB)')

File: scripts/og/generate-og.js
const { createCanvas } = require('canvas');
const fs = require('fs');

const W = 1200, H = 630;
const canvas = createCanvas(W, H);
const ctx = canvas.getContext('2d');

// Background
const grad = ctx.createLinearGradient(0, 0, W, H);
grad.addColorStop(0, '#0f172a');
grad.addColorStop(1, '#1e293b');
ctx.fillStyle = grad;
ctx.fillRect(0, 0, W, H);

// Grid dots
ctx.fillStyle = '#334155';
ctx.globalAlpha = 0.3;
for (let x = 60; x < W; x += 120)
  for (let y = 60; y < H; y += 120) {
    ctx.beginPath();
    ctx.arc(x, y, 2, 0, Math.PI * 2);
    ctx.fill();
  }
ctx.globalAlpha = 1;

// Decorative shapes
ctx.strokeStyle = '#08adff';
ctx.globalAlpha = 0.15;
ctx.lineWidth = 1;
roundRect(ctx, 80, 80, 80, 80, 16);
ctx.stroke();
roundRect(ctx, 1040, 470, 80, 80, 16);
ctx.stroke();
ctx.globalAlpha = 0.1;
ctx.beginPath();
ctx.arc(1100, 160, 20, 0, Math.PI * 2);
ctx.stroke();
ctx.beginPath();
ctx.arc(100, 530, 14, 0, Math.PI * 2);
ctx.stroke();
ctx.globalAlpha = 1;

// Logo text - "PicEte"
ctx.font = 'bold 60px system-ui, -apple-system, sans-serif';
ctx.textAlign = 'center';
ctx.textBaseline = 'bottom';
ctx.fillStyle = '#e2e8f0';
ctx.fillText('Pic', 530, 290);
ctx.fillStyle = '#08adff';
ctx.fillText('Ete', 670, 290);

// Tagline
ctx.font = '28px system-ui, -apple-system, sans-serif';
ctx.textBaseline = 'top';
ctx.fillStyle = '#94a3b8';
ctx.fillText('Free Online Image Processing Tools', 600, 340);

// Feature pills
const pills = [
  { label: 'Format Converter', x: 415 },
  { label: 'Image Resizer', x: 600 },
  { label: 'Compressor', x: 785 },
];
const pillY = 400;
for (const p of pills) {
  ctx.fillStyle = '#1e293b';
  ctx.strokeStyle = '#334155';
  ctx.lineWidth = 1;
  roundRect(ctx, p.x - 60, pillY, 120, 30, 15);
  ctx.fill();
  ctx.stroke();
  ctx.fillStyle = '#cbd5e1';
  ctx.font = '14px system-ui, -apple-system, sans-serif';
  ctx.textBaseline = 'middle';
  ctx.fillText(p.label, p.x, pillY + 15);
}

// URL
ctx.fillStyle = '#08adff';
ctx.font = 'bold 17px system-ui, -apple-system, sans-serif';
ctx.textBaseline = 'top';
ctx.fillText('picete.com', 600, 490);

function roundRect(ctx, x, y, w, h, r) {
  ctx.beginPath();
  ctx.moveTo(x + r, y);
  ctx.lineTo(x + w - r, y);
  ctx.quadraticCurveTo(x + w, y, x + w, y + r);
  ctx.lineTo(x + w, y + h - r);
  ctx.quadraticCurveTo(x + w, y + h, x + w - r, y + h);
  ctx.lineTo(x + r, y + h);
  ctx.quadraticCurveTo(x, y + h, x, y + h - r);
  ctx.lineTo(x, y + r);
  ctx.quadraticCurveTo(x, y, x + r, y);
  ctx.closePath();
}

const buf = canvas.toBuffer('image/png');
fs.writeFileSync('/home/wu/picete-site/images/og-image.png', buf);
console.log(`OG Image created: ${(buf.length / 1024).toFixed(1)} KB`);
